static/app/views/issueList/overview.actions.spec.tsx

```diff
@@ -321,4 +322,115 @@ describe('IssueListOverview (actions)', function () {
+        method: 'PUT',
+      });
+
+      render(<IssueListOverview {...defaultProps} />, {organization});
+
+      const groups = await screen.findAllByTestId('group');
+
+      await userEvent.click(
+        within(groups[0]).getByRole('checkbox', {name: /select issue/i})
+      );
+
+      expect(screen.getByText('Medium priority issue')).toBeInTheDocument();
+
+      // After action, will refetch so need to mock that response
+      MockApiClient.addMockResponse({
+        url: '/organizations/org-slug/issues/',
+        body: [highPriorityGroup],
+        headers: {Link: DEFAULT_LINKS_HEADER},
+      });
+
+      await userEvent.click(screen.getByRole('button', {name: /more issue actions/i}));
+      await userEvent.hover(screen.getByRole('menuitemradio', {name: /priority/i}));
+      await userEvent.click(screen.getByRole('menuitemradio', {name: /low/i}));
+
+      await waitFor(() => {
+        expect(updateIssueMock).toHaveBeenCalledWith(
+          '/organizations/org-slug/issues/',
+          expect.objectContaining({
+            query: expect.objectContaining({id: ['1']}),
+            data: {priority: PriorityLevel.LOW},
+          })
+        );
+
+        expect(screen.queryByText('Medium priority issue')).not.toBeInTheDocument();
+      });
+    });
+
+    it('does not remove issues after reprioritizing (when query includes all priorities)', async function () {
+      const updateIssueMock = MockApiClient.addMockResponse({
+        url: '/organizations/org-slug/issues/',
+        method: 'PUT',
+      });
+
+      render(
+        <IssueListOverview
+          {...defaultProps}
+          {...{
+            ...RouteComponentPropsFixture({
+              location: LocationFixture({
+                query: {query: 'is:unresolved'},
+              }),
+              params: {
+                orgId: organization.slug,
+                projectId: project.slug,
+                searchId: undefined,
+              },
+            }),
+          }}
+        />,
+        {organization}
+      );
+
+      const groups = await screen.findAllByTestId('group');
+
+      await userEvent.click(
+        within(groups[0]).getByRole('checkbox', {name: /select issue/i})
+      );
+
+      expect(screen.getByText('Medium priority issue')).toBeInTheDocument();
+
+      await userEvent.click(screen.getByRole('button', {name: /more issue actions/i}));
+      await userEvent.hover(screen.getByRole('menuitemradio', {name: /priority/i}));
+      await userEvent.click(screen.getByRole('menuitemradio', {name: /low/i}));
+
+      await waitFor(() => {
+        expect(updateIssueMock).toHaveBeenCalledWith(
+          '/organizations/org-slug/issues/',
+          expect.objectContaining({
+            query: expect.objectContaining({id: ['1']}),
+            data: {priority: PriorityLevel.LOW},
+          })
+        );
+
+        expect(screen.getByText('Medium priority issue')).toBeInTheDocument();
+      });
```

Comment:
RTL says best practice is to have only one thing in the waitFor or to use findBy if possible. you might be able to use find by on `screen.getByText('Medium priority issue')` and have no waitFor